Autopilot/Inc/gps.hpp

```diff
@@ -49,51 +58,74 @@ class Gps
         virtual void GetResult(GpsData_t *Data) = 0;
 };
 
-class NEOM8 : public Gps 
+class NEOM8 : public Gps
 {
 	public:
 		NEOM8(const NEOM8*) = delete;
 		static NEOM8* GetInstance();
 
 		/**
-         * Triggers interrupt for new GPS measurement - stores raw data in variables and returns right away
-         * */
+		 * Triggers interrupt for new GPS measurement - stores raw data in variables and returns right away
+		 * */
 		void BeginMeasuring() {}; //No need to call this to get data. Just call GetResult()
 
 		 /**GetResult should:
-         * 1. Reset dataIsNew flag
-         * 2. Transfers raw data from variables to struct
-         * 3. Updates utcTime and status values in struct as well
-         * */
+		 * 1. Reset dataIsNew flag
+		 * 2. Transfers raw data from variables to struct
+		 * 3. Updates utcTime and status values in struct as well
+		 * */
 		void GetResult(GpsData_t *Data);
 
+		/**
+		 * Returns the buffer used to receive GPS UART signals
+		 */
+		uint8_t* get_byte_collection_buffer();
+
+		/**
+		 * Goes through byte_collection_buffer, finds the messages we care about, and calls the appropriate parsing functions
+		 */
+		void parse_gpsData();
+
 	private:
 		//Constructor
 		NEOM8();
 
 		//Static instance
-		static NEOM8* gps_Instance;
-
-		//Variables
-		bool isDataNew = false;
-		bool ggaDataNew = false;
-		bool vtgDataNew = false;
-		long double measuredLatitude, measuredLongitude;  // 8 Bytes
-		float measuredUtcTime;     // 4 Bytes. Time in seconds since 00:00 (midnight)
-		float measuredGroundSpeed; // in m/s
-		int measuredAltitude; // in m
-		uint16_t measuredHeading; // in degrees. Should be between 0-360 at all times, but using integer just in case
-		uint8_t measuredNumSatellites;  // 1 Byte
-		uint8_t gpsFixStatus;
-		bool dataAvailable = false;
-
-		//Methods (Some code was transferred from PICpilot [https://github.com/UWARG/PICpilot])
-		bool is_check_sum_valid(char *);
-		uint8_t uint8_to_hex(unsigned int);
-		int ascii_to_hex(unsigned int);
-		void parse_gps_data();
-		void parse_vtg();
-		void parse_gga();
+		static NEOM8* gpsInstance;
+
+		// Buffers
+		uint8_t gga_buffer[GPS_UART_BUFFER_SIZE]; //buffer for pasing gga (positional packets)
+		uint8_t vtg_buffer[GPS_UART_BUFFER_SIZE]; //buffer for parsing vtg packets (velocity packets)
```

Comment:
ahahaha 3.2KB of buffers. Nice. All good, but hillarious

Reply:
We can always download more RAM. Also, if we can reduce the message size through actually configuring the sensor then we can reduce the buffer size significantly